src/main/java/com/redhat/vertx/pipeline/StepDependencyNotMetException.java

```diff
@@ -1,6 +1,6 @@
 package com.redhat.vertx.pipeline;
 
-public class StepDependencyNotMetException extends Exception {
+public class StepDependencyNotMetException extends Exception implements PotentiallyRecoverableException {
```

Comment:
This still seems odd to me. I would expect PotentiallyRecoverableException to subclass runtime (so we don't have to put it in a catch block) and then StepDependencyNotMetException to extend that.